### PR TITLE
feat: add support for angular two way data binding

### PR DESCRIPTION
**Pull request by fifoosid**

### CODE REVIEW

**Comment by ilhan007** [comment]
@fifoosid I have the context and I agree with the changes, but if there should be comments that it's done to enable two-way data binding, please add that it is for Angular

File: packages/main/src/DatePicker.js [hunk 1]
```diff
@@ -327,6 +327,8 @@ class DatePicker extends UI5Element {
 
 		this.value = nextValue;
 		this.fireEvent("change", { value: nextValue, valid: isValid });
+		// Two way data binding
+		this.fireEvent("value-changed", { value: nextValue, valid: isValid });
```
Comment (vladitasev) [inline packages/main/src/DatePicker.js]: Please check no-conflict mode (should not fire ui5- for value-changed)

File: packages/base/src/config/NoConflict.js [hunk 2]
```diff
@@ -1,9 +1,17 @@
 import { getNoConflict as getConfiguredNoConflict } from "../InitialConfiguration.js";
 
+const twoWayDataBindingEvents = [
```
Comment (ilhan007) [inline packages/base/src/config/NoConflict.js]: twoWayDataBindingEvents too specific, `excludeList` or something similar
isTwoWayDataBindingEvent as well, `skipPrefix`, `isEventExcluded` or similar

File: packages/base/src/config/NoConflict.js [hunk 3]
```diff
@@ -1,9 +1,37 @@
...
+
 let noConflict = getConfiguredNoConflict();
 
-const getNoConflict = () => {
-	return noConflict;
+const shouldNotFireOriginalEvent = eventName => {
+	return !(noConflict.events && noConflict.events.includes && noConflict.events.includes(eventName));
+};
+
+const getNoConflict = eventName => {
```
Comment (ilhan007) [inline packages/base/src/config/NoConflict.js]: Leave getNoConflict unchanged, export skipOriginalEvent and use it in UI5Element

**Comment by fifoosid** [comment]
Fixes #697 and #651

**Pull request merged**